servant-docs/src/Servant/Docs/Internal.hs

```diff
@@ -161,6 +161,22 @@ data DocQueryParam = DocQueryParam
   , _paramKind   :: ParamKind
   } deriving (Eq, Ord, Show)
 
+-- | A type to represent fragment. Holds the name of the fragment and its description.
+--
+-- Write a 'ToFragment' instance for your fragment types.
+data DocFragment = DocFragment
+  { _fragSymbol :: String -- type supplied
+  , _fragDesc   :: String -- user supplied
+  } deriving (Eq, Ord, Show)
+
+-- | Combine two Fragments, we can't make a monoid because merging Fragments breaks
+-- the laws.
+--
+-- As such, we invent a non-commutative, left associative operation
+-- 'combineFragment' to mush two together taking the info from the very left.
+combineFragment :: Maybe DocFragment -> Maybe DocFragment -> Maybe DocFragment
+mdocFragment `combineFragment` _ = mdocFragment
```

Comment:
1. Picking first occurrence is a valid monoid (see `First` in `Data.Monoid`).
2. I think ``Nothing `combineFragment` Just doc`` should be `Just doc`: if there was no fragment on the left, we should pick the fragment from the right.

```suggestion
-- | There should be at most one `Fragment` per API endpoint.
-- So here we are keeping the first occurrence.
combineFragment :: Maybe DocFragment -> Maybe DocFragment -> Maybe DocFragment
Nothing `combineFragment` mdocFragment = mdocFragment
Just docFragment `combineFragment` _ = Just docFragment
```

Reply:
Fixed.